### PR TITLE
Update modules chapter of tour

### LINKED ISSUE #473: package member naming rules unclear?
On this page
https://ceylon-lang.org/documentation/1.3/tour/modules/

under the headline "Source files and packages"

it is said that:
"There's no package statement in Ceylon source files. The compiler determines the package and module to which a toplevel program element belongs by the location of the source file in which it is declared."

Then an (interesting!) example is given:
"For example, if source is a source directory, and if a class named Hello is defined in the file source/org/jboss/hello/Hello.ceylon, then Hello belongs to the package org.jboss.hello."

This example leaves one question open:
Why was the package member named Hello?
Was it because the *file* was named Hello.ceylon?
Or because a class in the file was named Hello?
And what happens if there is no correspondence between file name and class name?

Now that I'm thinking more about it, I can kinda figure out that it must be the file name that determines the name of the package member. But it would still be nice having it stated explicit :)

### PR DESCRIPTION
**Pull request by lucaswerkmeister**
The introductory paragraph of the “Packages and imports” section dealt mostly with packages, not with imports, and went into a lot of detail. It has been updated to give a more high-level overview of both aspects, and leave the detail to its subsections.

The introduction to packages did not make it clear that the file name has no significance despite the convention to name files after their content. It has been rewritten to make the relation between declarations, source files, and packages more clear. The example has been moved to a separate paragraph and expanded.

The section on imports now explains the import syntax a bit more. The import examples have been expanded to include the example from the previous section, partially merged, and comments have been added.

Fixes #473.

---

@loldrup what do you think? (I’m afraid GitHub won’t let me add you as reviewer.)

### CODE REVIEW

**Comment by loldrup** [comment]
looks good! :)

I have two comments:


------


"The file name itself has no significance to the compiler,
but conventionally follows the content
of the source file."

Well, the compiler is a big thing, with many concerns. How about being more
explicit:

"In Ceylon we conventionally place every class in its own file,
with the same name as the class it contains. However, the file name
itself has no influence on the package name."


-------


"The file could also instead be
`source/org/jboss/hello/World.ceylon`;"

Indeed it could, but you forgot to state why this example is interesting.
How about:

"If the source file had instead been
`source/org/jboss/hello/World.ceylon`
the pckage name would still be the same."

(notice that I changed the semicolon to a period)

and then:

"On the other hand, the path
`source/org/jboss/world/Hello.ceylon`
would place the file in the
`org.jboss.world` package."

(nothing changed here)


-------

regards
Jon Loldrup

Mvh. Jon Loldrup

**Comment by gavinking** [comment]
I don't like some of the changes in this commit. It seems that some information has even been removed.

File: documentation/1.3/tour/modules.md [hunk 1]
```diff
@@ -18,23 +18,23 @@ We're going to learn about *packages* and *modules*.
 ## Packages and imports
 
 Every source file&mdash;along with everything declared in the 
-file&mdash;belongs to a package. Even source files that don't belong to a 
-named package are considered to belong to the "default" package. It's the 
-location of a source file within a source directory that determines which 
-package a source file belongs to.
```
Comment (gavinking) [inline documentation/1.3/tour/modules.md]: > Even source files that don't belong to a 
 -named package are considered to belong to the "default" package.

Why would you remove this information?

File: documentation/1.3/tour/modules.md [hunk 2]
```diff
@@ -18,23 +18,23 @@ We're going to learn about *packages* and *modules*.
...
 
 There's no `package` statement in Ceylon source files. The compiler determines 
 the package and module to which a toplevel program element belongs by the 
-location of the source file in which it is declared. For example, if `source` 
-is a source directory, and if a class named `Hello` is defined in the file 
-`source/org/jboss/hello/Hello.ceylon`, then `Hello` belongs to the package 
-`org.jboss.hello`.
+directory path of the source file in which it is declared. The file name itself 
+has no significance to the compiler, but conventionally follows the content 
+of the source file.
```
Comment (gavinking) [inline documentation/1.3/tour/modules.md]: "directory path of the source file" is clearer than "location of the source file"? I don't see why. How is "directory path" different to "location"?

Comment (gavinking) [inline documentation/1.3/tour/modules.md]: How is "the name of the source file itself is not significant" harder to understand than "the file name itself has no significance to the compiler, but conventionally follows the content of the source file"?

File: documentation/1.3/tour/modules.md [hunk 3]
```diff
@@ -18,23 +18,23 @@ We're going to learn about *packages* and *modules*.
...
 
-Note that the name of the source file itself is not significant, as long as
-it has the extension `.ceylon`. It's only the directory in which the source 
-file is located that matters to the compiler.
+For example, if `source` is a source directory, then a class named `Hello` 
+in the package `org.jboss.hello` would be defined in the file 
+`source/org/jboss/hello/Hello.ceylon`. The file could also instead be 
+`source/org/jboss/hello/World.ceylon`; on the other hand, the path 
+`source/org/jboss/world/Hello.ceylon` would place the file in the 
+`org.jboss.world` package.
```
Comment (gavinking) [inline documentation/1.3/tour/modules.md]: Seriously? This is extremely long-winded and confusing.

File: documentation/1.3/tour/modules.md [hunk 4]
```diff
@@ -43,30 +43,27 @@ another package, it must explicitly import that program element. Ceylon,
 unlike Java, does not support the use of qualified names within the source 
 file. We can't write `org.jboss.hello.Hello` in Ceylon.
 
-The syntax of the `import` statement is slightly different to Java. To import 
-a program element, we write:
+The syntax of the `import` statement is slightly different to Java. 
+To import program elements, we write the name of the package 
+and then list the program elements we want to import:
```
Comment (gavinking) [inline documentation/1.3/tour/modules.md]: Should be "To import _one or more_ program elements yada yada yada".

**Comment by gavinking** [comment]
Overall, I disagree that this commit improved the original text. Sorry.

**Comment by FroMage** [comment]
This PR has been closed as a result of moving Ceylon to the Eclipse Foundation, not as a judgement of its value. We are unfortunately unable to restore it ourselves, but if you feel this PR should be reopened, please rebase your work on the current master and create a new PR. Thanks, and sorry for the inconvenience.

**Pull request closed without merging**